### PR TITLE
fix(pgwire): fixed incorrect NULL type handing causing SEGFAULT in some cases

### PR DESCRIPTION
**Pull request by ideoma**
- upgraded PG driver (used by tests), assumption that old driver caused issues for CI when generating code coverage reports
- fixed hack, `'*!*table'` for generating no-rowid factories (useful for tests)

The crash arose from assumption that we can call getStr() method on NULL-types columns, which is not the case everywhere. In tests that used NULL type the column was always NullConstant, which did indeed implement getStr(). Crash came from RecordChain, which did not follow this convention.

The fix is to treat NULL columns as such and not attempt to call getStr() on them. This applies to both HTTP (JSON) and PG-wire.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 1 / 66 (01.52%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerConfigurationException.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|0|37|00.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|1|27|03.70%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 0 / 6 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|5|00.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|0|1|00.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

coverage check fail. please retry. :fearful:

[Please let me know](https://github.com/naver/cover-checker/issues/new) when error again.

com.naver.nid.cover.parser.coverage.exception.ParseException: java.io.FileNotFoundException: /home/vsts/work/1/jacoco-aggregate/jacoco.xml (No such file or directory)

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 0 / 6 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|5|00.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|0|1|00.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 0 / 6 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|5|00.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|0|1|00.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 1 / 6 (16.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|5|00.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 0 / 6 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|5|00.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|0|1|00.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 11 / 43 (25.58%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpContextConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|0|5|00.00%|
|:large_blue_circle:|io/questdb/std/GenericLexer.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|0|6|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|0|17|00.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cairo/RecordChain.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/DataFrameRecordCursorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 37 / 43 (86.05%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|5|00.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpContextConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/DeferredSingleSymbolFilterDataFrameRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/RecordChain.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/DataFrameRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/GenericLexer.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|11|11|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 14 / 43 (32.56%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpContextConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/ServerMain.java|0|5|00.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|0|6|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|0|17|00.00%|
|:large_blue_circle:|io/questdb/cairo/RecordChain.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/DataFrameRecordCursorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/std/GenericLexer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|8|8|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 37 / 43 (86.05%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|5|00.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessorState.java|16|17|94.12%|
|:large_blue_circle:|io/questdb/cutlass/http/DefaultHttpContextConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/RecordChain.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/DataFrameRecordCursorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/std/GenericLexer.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/log/Logger.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/pgwire/PGConnectionContext.java|6|6|100.00%|

File: core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java [hunk 1]
```diff
@@ -1121,11 +1121,13 @@ public void testAllParamsHex() throws Exception {
     }
 
     @Test
+    @Ignore
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java:1124]: What's the reason to ignore this test and the following one?

Comment (bluestreak01) [reply core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java:1124]: this is Alex hunting for code that broke test coverage

File: core/src/main/java/io/questdb/log/Logger.java [hunk 2]
```diff
@@ -358,7 +358,7 @@ private LogRecord next(Sequence seq, RingQueue<LogRecordSink> ring, int level) {
             return NullLogRecord.INSTANCE;
         }
 
-        final long cursor = seq.next();
+        final long cursor = seq.nextBully();
```
[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/main/java/io/questdb/log/Logger.java]: removed in https://github.com/questdb/questdb/pull/2087/commits/545b3b13c97c436c5f0c001453940221dbf6a0e3

**Review by puzpuzpuz — APPROVED** [review]

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**